### PR TITLE
Queue M->S messages in a per-room queue

### LINKED ISSUE #322: S-M edits broken in 1.0.2
I saw an edit made from slack and it doesn't come through propely onto matrix-side. With riot-web 1.5.3 it seems to show as fallback, on riot-android only shows fallback (by natuter), but most for example riotX does not show _anything_, like nothing was ever edited, thus no real edit nor fallback edit.. I suspect something is broken on appservice-slack relating edits this S>M.

All self-hosted:
Synapse 1.5.1
appservice-slack 1.0.2
riot-web/desktop 1.5.3
riot-android 0,9,8
riotX 0.7.0

Screenshots:
riot-web:
riot-web
riotX
riotX
riot-android:
riot-andoid

ADD:
Message sources, Original one:
```
{
  "type": "m.room.message",
  "sender": "@slack_helsinkihacklab_<redacted>:hacklab.fi",
  "content": {
    "body": "Turvaominaisuus, ei saa ainakaan molempia käsiä liikkuvien osien cäliin!",
    "format": "org.matrix.custom.html",
    "formatted_body": "Turvaominaisuus, ei saa ainakaan molempia käsiä liikkuvien osien cäliin!",
    "msgtype": "m.text"
  },
  "event_id": "$15737336231061FYGFD:hacklab.fi",
  "origin_server_ts": 1573733623423,
  "unsigned": {
    "age": 1125
  },
  "room_id": "!MjVdMmsFAxVzHUZLiV:matrix.org"
}
```

And from the edit:
```
{
  "type": "m.room.message",
  "sender": "@slack_helsinkihacklab_<redacted>:hacklab.fi",
  "content": {
    "body": "(edited) ... osien  cäliin!  => ... osien  väliin!  ",
    "format": "org.matrix.custom.html",
    "formatted_body": "<i>(edited)</i> ... osien  <font color=\"red\"> cäliin! </font>  =&gt; ... osien <font color=\"green\"> väliin! </font> ",
    "m.new_content": {
      "body": "Turvaominaisuus, ei saa ainakaan molempia käsiä liikkuvien osien väliin!",
      "format": "org.matrix.custom.html",
      "formatted_body": "Turvaominaisuus, ei saa ainakaan molempia käsiä liikkuvien osien väliin!",
      "msgtype": "m.text"
    },
    "m.relates_to": {
      "rel_type": "m.replace"
    },
    "msgtype": "m.text"
  },
  "event_id": "$15737336241062ZmRbl:hacklab.fi",
  "origin_server_ts": 1573733624638,
  "unsigned": {
    "age": 172
  },
  "room_id": "!MjVdMmsFAxVzHUZLiV:matrix.org"
}
```

### PR DESCRIPTION
**Pull request by Half-Shot**
Fixes #322

### CODE REVIEW

**Comment by Half-Shot** [comment]
(Sorry bruno)

File: src/BridgedRoom.ts [hunk 1]
```diff
@@ -592,6 +596,12 @@ export class BridgedRoom {
             `<font color="green"> ${curr} </font> ${after}`;
             const prevEvent = await this.main.datastore.getEventBySlackId(channelId, message.previous_message!.ts);
 
+            if (!prevEvent) {
```
Comment (benparsons) [inline src/BridgedRoom.ts]: this is a noop? is prevEvent ever read?

Comment (Half-Shot) [reply src/BridgedRoom.ts]: I think this was some mistaken code.

Comment (benparsons) [reply src/BridgedRoom.ts]: So we'll remove it?

File: src/BridgedRoom.ts [hunk 2]
```diff
@@ -474,7 +475,10 @@ export class BridgedRoom {
             const ghost = await this.main.getGhostForSlackMessage(message, this.slackTeamId!);
             await ghost.update(message, this);
             await ghost.cancelTyping(this.MatrixRoomId); // If they were typing, stop them from doing that.
-            return await this.handleSlackMessage(message, ghost, content);
+            this.slackSendLock = this.slackSendLock.finally(async () => {
```
Comment (auscompgeek) [inline src/BridgedRoom.ts]: Does this leak promises that have finished?

Comment (Half-Shot) [reply src/BridgedRoom.ts]: The following line should take care of it

**Pull request merged**